kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstance.java

```diff
@@ -120,6 +153,50 @@ public void onDisabled() {
         updateState(State::onDisabled);
     }
 
+    private ValidationResult validateContainerImageSignature(ContainerInstanceOptions configuration) {
+
+        if (Objects.isNull(this.availableContainerSignatureValidationService)
+                || this.availableContainerSignatureValidationService.isEmpty()) {
+            logger.warn("No container signature validation service available. Signature validation failed.");
+            return FAILED_VALIDATION;
+        }
+
+        Optional<String> optTrustAnchor = configuration.getSignatureTrustAnchor();
+        if (!optTrustAnchor.isPresent() || optTrustAnchor.get().isEmpty()) {
+            logger.warn("No trust anchor available. Signature validation failed.");
```

Comment:
`!optTrustAnchor.isPresent()` will never be true here

Reply:
Actually no, this code is reachable: if someone calls the `validateContainerImageSignature` without checking the trust anchor presence beforehand this branch can be traversed.

Reply:
But the only place where you call it is in the updated method, and it is private

Reply:
Yes but what if, years down the line, someone else needs to use this method and doesn't know about the requirement? It's a simple sanity check to prevent this method to be misused.

In any case, I can't remove it because otherwise Sonar will scream bloody murder at me:
![image](https://github.com/eclipse/kura/assets/22748355/ef9a6196-0bdb-42a8-adea-9014c35b4521)

Reply:
I think that we should change the method signature to have the already unwrapped parameters, leaving the unwrapping of `ContainerInstanceOptions` in the update method. Maybe overloading the method for reusable code.

```java
validateContainerImageSignature(String trustAnchor,  boolean verifyInTransparencyLog, String imageName, String imageTag) {
...
```

What do you think?

Reply:
I kinda disagree.

The purpose of the `validateContainerImageSignature` method is to:
- unwrap the `ContainerInstanceOptions` for the verification process (which involves checking for valid data)
- pass these data to the correct method exposed by the `ContainerSignatureValidationService`

the `updated` method should not know what the signature validation needs. It just calls the high level methods handling the container stuff (like the `onConfigurationUpdated`).

Also I don't think removing the `if` statement is needed: it serves as documentation as it makes clear what that method needs to work.

Reply:
Trying to convince you... I think that the update should call the correct methods based on the configuration provided, hence it should be the place where all the ifs and unwrapping of configurations happen.

`validateContainerImageSignature` should be renamed into `unwrapPropertiesAndValidateContainerImageSignature` if you proceed in that way. Now the method does not serve a single purpose. Moreover, a method called `validateX` should fail when inputs are wrong, not return a failed validation status.

Reply:
I understand what you're saying but I think you're looking at it from the wrong perspective.

> validateContainerImageSignature should be renamed into unwrapPropertiesAndValidateContainerImageSignature if you proceed in that way.

Going with this principle the `updated` method name would be *way* longer :D ... but looking at it *top-down* it does just that: given the new configuration, it updates the component.

> The purpose of the validateContainerImageSignature method is to:
>
> - unwrap the ContainerInstanceOptions for the verification process (which involves checking for valid data)
> - pass these data to the correct method exposed by the ContainerSignatureValidationService

Here I should have explained myself better: my purpose with the `validateContainerImageSignature` method is to have something **self-contained**. By proceeding you're way we would be splitting the method unnecessarily.

Again, looking at it *top-down*, the `validateContainerImageSignature` method does only one thing: given the container configuration it validates the signature. It hides the complexity inside it because, at the `updated` level of abstraction we don't care what happens inside.

To sum everything up:
- The highlighted code *is reachable*
- The check performed *is necessary* both from the CI perspective (Sonar whines about it) and from the readability perspective (it's clear what the method expects i.e. the preconditions)
- The purpose of the method is to have something *self-contained* and therefore extracting the unwrapping of the option in the `updated` method is working against that principle. Furthermore having the unwrapping at the `updated` level is mixing the abstraction layers.

Reply:
Ok